CONTRIBUTING.md

```diff
@@ -32,3 +32,65 @@ From the Chisel root directory, run:
 # Reformat mill build files
 ./mill --meta-level 1 mill.scalalib.scalafmt.ScalafmtModule/reformatAll sources
 ```
+
+---
+
+## Internal Documentation
+
+This section describes the internal Chisel components for contributors.
+
+### CIRCT Converter
+
+There is a highly experimental component CIRCT Converter (a.k.a. Panama Converter). It is powered by [Java's Panama framework](https://openjdk.org/projects/panama/) to interact directly with CIRCT by calling the C-APIs of MLIR and CIRCT directly from Scala, enabling seamless emitting Chisel IR to CIRCT FIRRTL Dialect IR (no serialization and deserialization for FIRRTL), flexible executing Passes with PassManager, lowering to / exporting SystemVerilog, accesing OM data, and more.
+
+#### Directory `circtpanamabinding`
+
+Here defines the needed CIRCT items that will be processed by the Panama framework's [jextract](https://github.com/openjdk/jextract) tool for codegen FFI Java code to use in Scala.
+
+#### Directory `panamalib`
+
+It provides a type-safe wrapper for the FFI code generated by jextract.
+
+#### File `PanamaCIRCTConverter.scala`
+
+Here is the implementation of how each Chisel IR will be emitted to the CIRCT.
+
+It needs to be highly synchronized with CIRCT upstream. When updating it, you can refer to [CIRCT's FIRRTL Dialect documentation](https://circt.llvm.org/docs/Dialects/FIRRTL/). Some dialect-specific types, conversions may require a specialized C-API function to return from CIRCT, in which case you can open a PR in CIRCT upstream to add it and use it here.
+
+#### File `PanamaCIRCTPassManager.scala`
+
+It provides a PassManager, which is used in place of the firtool cli.
+
+#### File `PanamaCIRCTOM.scala`
+
+After using PassManager to lowering the FIRRTL Dialect to the HW Dialect, you will be able to access the OM data in it.
+
+The data types of the OM are defined here, as well as the way to access them. This is an example of printing out all the `Top_Class` fields.
+
+```scala
+val pm = converter.passManager()
+assert(pm.populateFinalizeIR())
+assert(pm.run())
+
+val om = converter.om()
+val evaluator = om.evaluator()
+
+val top = evaluator.instantiate("Top_Class", Seq(om.newBasePathEmpty)).get
+top.foreachField((name, value) => println(s".$name => { ${value.toString} }"))
+```
+
+#### Directory `lit`
+
+We used llvm-lit and scala-cli to test CIRCT Converter.
+
+To run the test, make sure the environment variables are defined first:
+
+- `JAVA_HOME`: The directory where the Java runtime environment. The Panama framework requires Java 21 and higher.
+
+```
+mill -i lit[_].run
+```
+
+#### Debugging
+
+When interacting with CIRCT, it is easy to get a JVM crash, which is usually due to an unexpected error inside CIRCT (e.g. an assertion failure). Checking the call stack of the crash will help you to localize the bug.
```

Comment:
Please add more documentation for debugging, you can put the documentation on `docs/src/developers`, since this flow has been agnostic to all other developers. I think you need to provide a `scala-cli` as an example, you can copy from some thing from the lit folder. Please give a example how to configure the CICRT for debugging, including
1. how to recompile CIRCT to dynamic library to debug info(`-DCMAKE_BUILD_TYPE=RelWithDebInfo`);
2. how to generate the jextract header(point a link to [`common.sc`](https://github.com/chipsalliance/chisel/blob/4f392323e9160440961b9f06e383d3f2742d2f3e/common.sc#L173))
3. how to link the jvm program to the generated dynamic library(some JVM options added on the scala-cli, maybe the [`SmokeTest.sc`](https://github.com/chipsalliance/chisel/blob/main/lit/tests/SmokeTest.sc) is enough);
4. how to hook gdb to to dynamic library(this is really important, since for most of other developer doesn't know this flow, including myself);

Reply:
I don't use gdb to debug it either. All I need is just `println` :/